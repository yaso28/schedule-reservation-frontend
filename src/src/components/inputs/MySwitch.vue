<template>
  <MyFormGroup
    :label="label"
    :form="form"
    :field="field"
  >
    <MySwitchNoWrapper
      :form="form"
      :field="field"
    />
  </MyFormGroup>
</template>

<script>
export default {
  name: 'MySwitch',
  components: {
    MyFormGroup: require('./MyFormGroup').default,
    MySwitchNoWrapper: require('./MySwitchNoWrapper').default
  },
  mixins: [require('./input-mixin').default]
}
</script>
