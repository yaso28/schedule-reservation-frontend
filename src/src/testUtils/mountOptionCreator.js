import { createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import routes from '@/router/routes'
import Vuex from 'vuex'
import VueI18n from 'vue-i18n'
import i18nOptions from '@/i18n/options'
import VueMeta from 'vue-meta'
import CoreuiVue from '@coreui/vue'
import icons from '@/icons'
import services from '@/services'

class MountOptionCreator {
  create (storeModules = null) {
    const localVue = createLocalVue()

    // router
    localVue.use(VueRouter)
    const router = new VueRouter({
      mode: 'history',
      base: process.env.BASE_URL,
      routes
    })

    // i18n
    localVue.use(VueI18n)
    const i18n = new VueI18n(i18nOptions)

    // vue-meta
    localVue.use(VueMeta)

    // Core UI
    localVue.use(CoreuiVue)

    const option = {
      localVue,
      router,
      i18n,
      icons,
      provide: services
    }

    // store
    if (storeModules) {
      localVue.use(Vuex)
      option.store = new Vuex.Store({
        modules: storeModules
      })
    }

    return option
  }
}

export default new MountOptionCreator()
