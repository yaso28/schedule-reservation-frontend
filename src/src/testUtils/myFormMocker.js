import MyForm from '@/models/MyForm'
jest.mock('@/models/MyForm')

class MyFormMocker {
  createMockSingleField (field, initValue) {
    const input = {}
    input[field] = initValue
    return this.createMock(input)
  }

  createMock (input) {
    const mockForm = new MyForm(input)
    mockForm.values = input
    mockForm.errors = {}
    return mockForm
  }

  async setValue (wrapper, form, field, value) {
    await wrapper.setProps({
      form: Object.assign({}, form, {
        values: Object.assign({}, form.values, { [field]: value })
      })
    })
  }

  setValueNonProps (form, field, value) {
    form.values[field] = value
  }

  async setError (wrapper, form, field, errorMessage) {
    await wrapper.setProps({
      form: Object.assign({}, form, {
        errors: Object.assign({}, form.errors, { [field]: errorMessage })
      })
    })
  }
}

export default new MyFormMocker()
