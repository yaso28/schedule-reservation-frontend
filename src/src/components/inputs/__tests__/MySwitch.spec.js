import { mount } from '@vue/test-utils'
import mountOptionCreator from '@/testUtils/mountOptionCreator'
import MySwitch from '../MySwitch'
import myFormMocker from '@/testUtils/myFormMocker'

jest.mock('@/models/MyForm')

function getInputWrapper (wrapper) {
  return wrapper.find('input')
}

describe('MySwitch', () => {
  let form
  let field
  let wrapper
  beforeEach(() => {
    const label = 'サンプル'
    field = 'sample_field'
    form = myFormMocker.createMockSingleField(field, false)

    const mountOption = mountOptionCreator.create()
    mountOption.propsData = { label, form, field }
    wrapper = mount(MySwitch, mountOption)
  })

  afterEach(() => {
    jest.resetAllMocks()
  })

  describe('通常', () => {
    test('表示', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('バリデーションエラー', () => {
    beforeEach(async () => {
      await myFormMocker.setError(wrapper, form, field, 'エラーです')
    })

    test('表示', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('入力', () => {
    let inputWrapper
    beforeEach(() => {
      inputWrapper = getInputWrapper(wrapper)
    })

    describe('チェックON', () => {
      beforeEach(async () => {
        await myFormMocker.setValue(wrapper, form, field, false)
        await inputWrapper.setChecked(true)
      })

      test('MyForm.setValue()を呼出', () => {
        expect(form.setValue).toHaveBeenCalledWith(field, true)
      })
    })

    describe('チェックOFF', () => {
      beforeEach(async () => {
        await myFormMocker.setValue(wrapper, form, field, true)
        await inputWrapper.setChecked(false)
      })

      test('MyForm.setValue()を呼出', () => {
        expect(form.setValue).toHaveBeenCalledWith(field, false)
      })
    })
  })
})
