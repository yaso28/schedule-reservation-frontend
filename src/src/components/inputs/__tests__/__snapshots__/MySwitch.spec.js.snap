// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MySwitch バリデーションエラー 表示 1`] = `
<div
  class="form-row form-group"
>
  <label
    class="col-form-label col-sm-3"
    for=""
  >
    
    サンプル
  
  </label>
   
  <div
    class="col-sm-9"
  >
    <div
      class="my-input-wrapper is-invalid"
    >
      <label
        class="c-switch form-check-label c-switch-primary c-switch-label"
      >
        <input
          class="c-switch-input form-check-input"
          id="sample_field"
          type="checkbox"
        />
        <span
          class="c-switch-slider"
          data-checked="✓"
          data-unchecked="✕"
        />
      </label>
    </div>
     
    <div
      class="invalid-feedback"
    >
      
      エラーです
    
    </div>
  </div>
</div>
`;

exports[`MySwitch 通常 表示 1`] = `
<div
  class="form-row form-group"
>
  <label
    class="col-form-label col-sm-3"
    for=""
  >
    
    サンプル
  
  </label>
   
  <div
    class="col-sm-9"
  >
    <div
      class="my-input-wrapper"
    >
      <label
        class="c-switch form-check-label c-switch-primary c-switch-label"
      >
        <input
          class="c-switch-input form-check-input"
          id="sample_field"
          type="checkbox"
        />
        <span
          class="c-switch-slider"
          data-checked="✓"
          data-unchecked="✕"
        />
      </label>
    </div>
     
    <!---->
  </div>
</div>
`;
